src/replica_server/replication_lib/mutation.cpp

```diff
@@ -285,7 +286,18 @@ mutation_queue::mutation_queue(gpid gpid, int max_concurrent_op /*= 2*/, bool ba
 
 mutation_ptr mutation_queue::add_work(task_code code, dsn_message_t request, replica* r)
 {
-    // batch and add to work queue
+    task_spec* spec = task_spec::get(code);
+
+    // if not allow write batch, switch work queue
+    if (_pending_mutation && !spec->rpc_request_is_write_allow_batch)
```

Comment:
This is doable. Or, you can have your own options of "task-spec" array, as we have done in cases such as [profiler](https://github.com/Microsoft/rDSN/blob/master/src/plugins/tools.common/profiler.cpp#L86). In that way, we avoid changing the core code due to the requirement from certain plugins.